bulkcopy.go

```diff
@@ -0,0 +1,576 @@
+package mssql
+
+import (
+	"bytes"
+	_ "database/sql/driver"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"math"
+	"reflect"
+	"strings"
+	"time"
+	"golang.org/x/net/context" // use the "x/net/context" for backwards compatibility.
+)
+
+type MssqlBulk struct {
+	cn          *MssqlConn
+	metadata    []columnStruct
+	bulkColumns []columnStruct
+	columnsName []string
+	tablename   string
+	numRows     int
+
+	headerSent bool
+	Options    MssqlBulkOptions
+	Debug      bool
+}
+type MssqlBulkOptions struct {
+	CheckConstraints  bool
+	FireTriggers      bool
+	KeepNulls         bool
+	KilobytesPerBatch int
+	RowsPerBatch      int
+	Order             []string
+	Tablock           bool
+}
+
+type DataValue interface{}
+
+func (cn *MssqlConn) CreateBulk(table string, columns []string) (_ *MssqlBulk) {
+	b := MssqlBulk{cn: cn, tablename: table, headerSent: false, columnsName: columns}
+	b.Debug = false
+	return &b
+}
+
+func (b *MssqlBulk) sendBulkCommand() error {
+
+	//get table columns info
+	err := b.getMetadata()
+	if err != nil {
+		return err
+	}
+	//match the columns
+	if b.Debug {
```

Comment:
Please provide a function called (*MssqlBulk) dlogf which checks this conditional and prints to the defined logger.